devops/providers/azure-devops/templates/infrastructure/azure-pipeline-build-stage.yml

```diff
@@ -1,39 +1,82 @@
 parameters:
-  stageName: ''
   environment: ''
-  dependentStageName: ''
-  artifact: ''
+  configurationMatrix: []
 
-stages:
-- stage: ${{ parameters.stageName }}
-  dependsOn: ${{ parameters.dependentStageName }}
-  condition: succeeded('${{ parameters.dependentStageName }}')
-  jobs:
-  - job: Build
+jobs:
 
+- job: TemplateChangeDetection
+  displayName: Determine CI Targets to Run
+  steps:
+  - ${{ each config in parameters.configurationMatrix }}:
```

Comment:
how would i address not running this on each, or making this conditional?

Reply:
Can you elaborate on what you mean here? Do you mean to ask how this step could be omitted and we default to "run ci/cd on everything"?